src/test/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapperTest.java

```diff
@@ -0,0 +1,225 @@
+        requestChunk.clear();
+        writer.commit();
+      }
+    }
+  }
+
+  static class TestScriptPlugin extends Plugin implements ScriptPlugin {
+    @Override
+    public Iterable<ScriptEngine> getScriptEngines(List<ScriptContext<?>> contexts) {
+      return Arrays.asList(new TestScriptEngine());
+    }
+  }
+
+  public static class TestScriptEngine implements ScriptEngine {
+
+    @Override
+    public String getLang() {
+      return "test_lang";
+    }
+
+    @Override
+    public <T> T compile(String source, ScriptContext<T> context) {
+      ScoreScript.Factory factory =
+          ((params, docLookup) -> new TestScriptFactory(params, docLookup, source));
+      return context.factoryClazz.cast(factory);
+    }
+  }
+
+  static class TestScriptFactory extends ScoreScript.SegmentFactory {
+    private final Map<String, Object> params;
+    private final DocLookup docLookup;
+    private final String scriptId;
+
+    public TestScriptFactory(Map<String, Object> params, DocLookup docLookup, String scriptId) {
+      super(params, docLookup);
+      this.params = params;
+      this.docLookup = docLookup;
+      this.scriptId = scriptId;
+    }
+
+    @Override
+    public boolean needs_score() {
+      return true;
+    }
+
+    @Override
+    public DoubleValues newInstance(LeafReaderContext ctx, DoubleValues scores) {
+      return new DoubleScoreScript(params, docLookup, ctx, scores);
+    }
+
+    public DocLookup getDocLookup() {
+      return docLookup;
+    }
+  }
+
+  static class DoubleScoreScript extends ScoreScript {
+    public DoubleScoreScript(
+        Map<String, Object> params,
+        DocLookup docLookup,
+        LeafReaderContext context,
+        DoubleValues scores) {
+      super(params, docLookup, context, scores);
+    }
+
+    @Override
+    public double execute() {
+      return 2.4;
+    }
+  }
+
+  @Test
+  public void testFunctionScoreQueryField() throws IOException {
+    init(Collections.singletonList(new TestScriptPlugin()));
+    SearchRequest request =
+        SearchRequest.newBuilder()
+            .setIndexName(TEST_INDEX)
+            .setTopHits(NUM_DOCS)
+            .addAllRetrieveFields(RETRIEVE_LIST)
+            .setQuery(getQuery())
+            .build();
+    IndexState indexState = getGlobalState().getIndex(TEST_INDEX);
+    ShardState shardState = indexState.getShard(0);
+    SearcherTaxonomyManager.SearcherAndTaxonomy s = null;
+
+    try {
+      s = shardState.acquire();
+      SearchContext context =
+          SearchRequestProcessor.buildContextForRequest(request, indexState, shardState, s, null);
+      org.apache.lucene.queries.function.FunctionScoreQuery query =
+          (org.apache.lucene.queries.function.FunctionScoreQuery) context.getQuery();
+      assertNotNull(query);
+      TestScriptFactory source = (TestScriptFactory) query.getSource();
+      assertEquals(source.getDocLookup().getFieldDef("int_field").getName(), "int_field");
```

Comment:
This is an index field. It would be good to test accessing a virtual field specified in the query.